base/abstractarray.jl

```diff
@@ -1771,9 +1771,13 @@ typed_vcat(::Type{T}, A::AbstractVecOrMat...) where {T} = _typed_vcat(T, A)
 
 reduce(::typeof(vcat), A::AbstractVector{<:AbstractVecOrMat}) =
     _typed_vcat(mapreduce(eltype, promote_type, A), A)
+reduce(::typeof(vcat), A::Tuple{Vararg{AbstractVecOrMat}}) =
```

Comment:
Also avoid empty case here:
```suggestion
reduce(::typeof(vcat), A::Tuple{AbstractVecOrMat, Vararg{AbstractVecOrMat}}) =
```